### PR TITLE
chore(cli): implement env deployer that uploads artifacts

### PR DESCRIPTION
**Pull request by Lou1415926**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/deploy/env.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// UploadArtifacts uploads the deployment artifacts for the environment.
+func (d *envDeployer) UploadArtifacts() (map[string]string, error) {
+	envRegion := d.env.Region
+
+	resources, err := d.getAppRegionalResources()
+	if err != nil {
+		return nil, err
+	}
+	if resources.S3Bucket == "" {
+		log.Errorf("Cannot find the S3 artifact bucket in %s region created by app %s. The S3 bucket is necessary for many future operations. For example, when you need addons to your services.", envRegion, d.app.Name)
```
Comment (huanjani) [inline internal/pkg/cli/deploy/env.go]: Should we break this up into two or three lines, for width reasons?

Comment (huanjani) [inline internal/pkg/cli/deploy/env.go]: maybe?
```suggestion
		log.Errorf("Cannot find the S3 artifact bucket in region %s created by app %s. The S3 bucket is necessary for many future operations. For example, when you need addons to your services.", envRegion, d.app.Name)
```

File: internal/pkg/cli/deploy/env.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		return nil, fmt.Errorf("cannot find the S3 artifact bucket in %s region", envRegion)
```
Comment (huanjani) [inline internal/pkg/cli/deploy/env.go]: (same as above)
```suggestion
		return nil, fmt.Errorf("cannot find the S3 artifact bucket in region %s", envRegion)
```

File: internal/pkg/cli/deploy/env.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+func (d *envDeployer) getAppRegionalResources() (*stack.AppRegionalResources, error) {
+	if d.appRegionalResources != nil {
+		return d.appRegionalResources, nil
+	}
+	resources, err := d.appCFN.GetAppResourcesByRegion(d.app, d.env.Region)
+	if err != nil {
+		return nil, fmt.Errorf("get app resources in region %s: %w", d.env.Region, err)
+	}
+	return resources, nil
+}
```
Comment (huanjani) [inline internal/pkg/cli/deploy/env.go:65]: tiny nit: add extra line at end of this file and the test file?

File: internal/pkg/cli/deploy/env.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/cli/deploy/env.go]: Why do we need this log msg? I feel it is not a common error unless they remove the bucket manually?

File: internal/pkg/cli/deploy/env.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	"github.com/aws/copilot-cli/internal/pkg/aws/s3"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/copilot-cli/internal/pkg/manifest"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+)
+
+type appResourcesGetter interface {
+	GetAppResourcesByRegion(app *config.Application, region string) (*stack.AppRegionalResources, error)
+	GetRegionalAppResources(app *config.Application) ([]*stack.AppRegionalResources, error)
```
Comment (iamhopaul123) [inline internal/pkg/cli/deploy/env.go]: Are we using this?

**Review by huanjani — APPROVED** [review]
Just a few nits; otherwise, LGTM!

File: internal/pkg/cli/deploy/env.go (hunk 1 above)
Comment (Lou1415926) [reply internal/pkg/cli/deploy/env.go]: The log message was moved over from `cli/env_init.go` but sure we can remove it!

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/cli/deploy/env.go (hunk 3 above)
Comment (Lou1415926) [reply internal/pkg/cli/deploy/env.go:65]: I have it locally but after I pushed it got removed 🤔

**Pull request merged**